### ISSUE TITLE
Socket IO 1.0.* is not stable - Major UTF-8 issues due to engine.io version inconsitency  ("invalid continuation byte")

### ISSUE THREAD

**Issue by Cleod9** [socketio/socket.io#1744]
Hi there, I'm not sure if this issue would be more appropriate in the engine.io issue log, but since this issue bubbles up to socket.io I've decided to post it here. I'll try to make this as clear as I can since it took me some investigation to figure this out- 

To give you a quick overview, I'm running a server that can have several hundred users connected at once at any time with socket.io version 1.0.6. But I often receive this UTF error "invalid continuation byte" which causes the server to crash 1-2 times an hour. This disrupts service for my users so I decided to take matters into my own hands, and it appears there is a serious issue with engine-io-parser's versioning within the dependency tree. I'll explain:

First, please note that socket.io has engine.io version 1.3.1 as a dependency in its package.json
https://github.com/Automattic/socket.io/blob/1.0.6/package.json

Next, note that engine.io tagged at v1.3.1 has engine.io-parser: v1.0.6 listed as a dependency in its package.json. This is not desired for socket.io 1.0.6, as it is missing a crucial UTF-8 fix ("invalid continuation byte" error)
https://github.com/Automattic/engine.io/blob/1.3.1/package.json
(Example error: https://github.com/Automattic/engine.io/issues/264 )

But what we really want is engine.io-parser: "1.1.0"),
https://github.com/Automattic/engine.io/blob/15afd3e2c38ee03010d98852a570849e3a9deec2/package.json

So if you compare the above links you will see that the version was not updated in this package.json between engine.io's 1.3.1 tag and commit #15afd3e2c3. As a result, people who grab socket.io 1.0.6 are always going to receive the outdated engine.io-parser 1.0.6. Unless there is some serious compatibility issue between engine io parser 1.0.6 and 1.1.0, I highly recommend bumping up the versions of all of these libraries to get them back in sync with each other. I have tested the fix myself by hacking in engine.io parser 1.1.0 and I've already gone 24 hours without any app crashes from the UTF-8 bug.

So in summary, I recommend performing the following tasks in this order to resolve this:

-Bump engine.io to version 1.3.2 (Tagging the commit as 1.3.2 obviously)
-Bump socket.io to version 1.0.7 referencing engine.io version 1.3.2 as a dependency (and of course remember to tag this commit)

I would greatly appreciate it if you are able to verify these changes are valid make these quick updates. I'm sure it will save a lot of people trouble in the future :)

(And next time for engine.io, pleeeeease make sure commits with such important fixes get tagged and socket.io is updated accordingly. Since socket.io relies so much on the engine.io's sub modules, I would expect this library to be updated almost as frequently as its dependencies)
